src/ebay-textbox/textbox.tsx

```diff
@@ -56,34 +81,63 @@ const EbayTextbox: FC<EbayTextboxProps> = ({
         disabled: rest.disabled,
         label: floatingLabel,
         inputSize,
-        inputValue: controlledValue || value,
+        inputValue: controlledValue || inputValue,
         placeholder,
-        invalid
+        invalid,
+        onMount: onFloatingLabelInit
     })
 
     const handleFocus = (event?: FocusEvent<HTMLInputElement & HTMLTextAreaElement>) => {
-        onFocus(event)
+        onFocus(event, { value: event?.target?.value || defaultValue })
         onFloatingLabelFocus()
     }
 
     const handleBlur = (event: FocusEvent<HTMLInputElement & HTMLTextAreaElement>) => {
-        onBlur(event)
+        const newValue = event.target?.value
+        onBlur(event, { value: newValue })
+        if (newValue !== value) {
+            onChange(event, { value: newValue })
+            setValue(newValue)
+        }
```

Comment:
This can be dangerous by running onChange handlers inside an onBlur. The current `value` might not be the most updated value and triggers an unexpected onChange